### DISCUSSION TITLE
Liquid with transmission inside a bottle rendered transparent

### DISCUSSION THREAD

**Discussion by commit-master** [google/model-viewer#3910] [Q&A]
So I have this glb model that is a model with a colored liquid inside of it. It does not render at all on model-viewer but it does on other renderers like babylonJS for example.

See the two screenshots attached:

![model-viewer](https://user-images.githubusercontent.com/11718935/198641151-d60640c5-51d7-453b-8f71-58a24bd4ef00.png)

![babylonjs](https://user-images.githubusercontent.com/11718935/198641452-75bcaa90-36db-48d9-a021-9bed5dc24d9d.png)

Do you happen to have any idea on what could be causing it ? Is there a way to make it work (I was thinking detecting the material and modifying it via threejs?)

Thanks

**Comment by elalish** [comment]
I can't really say much without the actual GLB; can you attach it? I'd recommend looking for errors/warnings in our editor: https://modelviewer.dev/editor/

Do you know how this was exported and what extensions it uses? For colored liquid I would recommend the KHR_materials_volume extension. Still, multiple layers of transparency is always going to be tricky for a rasterizer.

**Comment by commit-master** [comment]
Hey @elalish  sorry for the late response I wanted to have all the elements before. No errors or warnings in your editor.

Here is the GLB (add to compress since github does not allow .glb files): 
[GINGKO_GLB.glb.zip](https://github.com/google/model-viewer/files/9919558/GINGKO_GLB.glb.zip)

I did use KHR_materials_volume I believe but it does not seem to do the trick..

**Reply by elalish** [reply]
If you look at the last tab of the editor is shows your glTF JSON. There's no KHR_materials_volume, only transmission. Still, I'm not sure that's the problem. Why do you have a separate bottle and label? The label should just be part of the bottle's texture; that might help a lot. I'm not totally sure how multiple layers of transmission materials are handled currently; might be worth making a test object to see. This could be a three.js bug.
